Support RISV64

### your feature request related to a problem? Please describe.

```sh
GOARCH=riscv64 go build ./cmd/golangci-lint
```

fails with:

```
/go/pkg/mod/github.com/prometheus/procfs@v0.1.3/cpuinfo.go:71:9: undefined: parseCPUInfo
```

### Describe the solution you'd like.

I expect golangci-lint to be buildable for riscv64 as it is for all other platforms.

There is the dependency `github.com/prometheus/procfs` which added support for riscv64 in `v0.6.0` (https://github.com/prometheus/procfs/commit/910e68572b35da779e4f84cfa5261b0a67532d05)

golangci-lint probably depends on many dependencies using versions pre-v0.6.0 of procfs so it's more of a long shot than something that can be fixed in a single PR I believe.

### Describe alternatives you've considered.

```sh
go mod edit -replace=github.com/prometheus/procfs=github.com/prometheus/procfs@v0.6.0
go mod tidy
```

Fixes the build, but probably in a risky-ish way I guess. Not a priority issue at all, but I thought it would be useful to track it.

### Additional context

_No response_

--- a/.goreleaser.yml
+++ b/.goreleaser.yml
@@ -22,6 +22,7 @@ builds:
       - s390x
       - mips64
       - mips64le
+      - riscv64
     goarm:
       - 6
       - 7

--- a/go.mod
+++ b/go.mod
@@ -58,6 +58,7 @@ require (
 	github.com/nishanths/predeclared v0.2.1
 	github.com/pkg/errors v0.9.1
 	github.com/polyfloyd/go-errorlint v0.0.0-20210510181950-ab96adb96fea
+	github.com/prometheus/procfs v0.6.0 // indirect
 	github.com/ryancurrah/gomodguard v1.2.2
 	github.com/ryanrolds/sqlclosecheck v0.3.0
 	github.com/sanposhiho/wastedassign/v2 v2.0.6

